packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -188,12 +249,35 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
       throw new Error('At least one of \'port\' or \'protocol\' is required');
     }
 
+    if (props.mutualAuthentication) {
+      const currentMode = props.mutualAuthentication.mutualAuthenticationMode;
+
+      if (currentMode === MutualAuthenticationMode.VERIFY && !props.mutualAuthentication.trustStore) {
```

Comment:
I think these two `if` blocks can be merged with something like this.
```
if props.mutualAuthentication.trustStore
  if currentMode !== MutualAuthenticationMode.VERIFY
    throw new Error ('trustStore can only be set when the mode is verify'
```

Reply:
Thank you. I've extracted the validation into a separate function to make it cleaner. 
What do you think of this approach?

https://github.com/aws/aws-cdk/pull/30784/files#diff-7bb91966b981c92a62f92075cd5bc4de58634ddcbe15dd492b0466f24b49383cR1041